server/src/test/java/org/elasticsearch/index/engine/LiveVersionMapTests.java

```diff
@@ -396,6 +397,50 @@ public void testPruneTombstonesWhileLocked() throws InterruptedException, IOExce
         assertEquals(0, map.getAllTombstones().size());
     }
 
+    public void testNeverLeaveStaleDeleteTombstone() throws Exception {
```

Comment:
can you explain why we need concurrency here? (i.e., why a single threaded test is not enough)

Reply:
A single threaded test is good enough ;)